core/services/ocr2/plugins/ocr2keeper/evm21/abi.go

```diff
@@ -13,23 +13,30 @@ import (
 	iregistry21 "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/i_keeper_registry_master_wrapper_2_1"
 )
 
-// enum UpkeepFailureReason is defined by https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/dev/automation/2_1/interfaces/AutomationRegistryInterface2_1.sol#L97
-// make sure failure reasons are in sync between contract and offchain enum
 const (
-	UPKEEP_FAILURE_REASON_NONE = iota
-	UPKEEP_FAILURE_REASON_UPKEEP_CANCELLED
-	UPKEEP_FAILURE_REASON_UPKEEP_PAUSED
-	UPKEEP_FAILURE_REASON_TARGET_CHECK_REVERTED
-	UPKEEP_FAILURE_REASON_UPKEEP_NOT_NEEDED
-	UPKEEP_FAILURE_REASON_PERFORM_DATA_EXCEEDS_LIMIT
-	UPKEEP_FAILURE_REASON_INSUFFICIENT_BALANCE
-	UPKEEP_FAILURE_REASON_MERCURY_CALLBACK_REVERTED
-	UPKEEP_FAILURE_REASON_REVERT_DATA_EXCEEDS_LIMIT
-	UPKEEP_FAILURE_REASON_REGISTRY_PAUSED
-
-	// Start of offchain failure types. All onchain failure reasons from
-	// contract should be put above
-	UPKEEP_FAILURE_REASON_MERCURY_ACCESS_NOT_ALLOWED
+	// upkeep failure
+	UpkeepFailureReasonNone = iota
+	UpkeepFailureReasonUpkeepCancelled
+	UpkeepFailureReasonUpkeepPaused
+	UpkeepFailureReasonTargetCheckReverted
+	UpkeepFailureReasonUpkeepNotNeeded
+	UpkeepFailureReasonPerformDataExceedsLimit
+	UpkeepFailureReasonInsufficientBalance
+	UpkeepFailureReasonMercuryCallbackReverted
+	UpkeepFailureReasonRevertDataExceedsLimit
+	UpkeepFailureReasonRegistryPaused
+	UpkeepFailureReasonMercuryAccessNotAllowed
+	UpkeepFailureReasonLogBlockNoLongerExists
+	UpkeepFailureReasonLogBlockInvalid
+	UpkeepFailureReasonTxHashNoLongerExists
+
+	// pipeline execution error
+	NoPipelineError     = 0
+	CheckBlockTooOld    = 1
+	CheckBlockInvalid   = 2
+	RpcFlakyFailure     = 3
+	MercuryFlakyFailure = 4
+	PackUnpackFailed    = 5
```

Comment:
nit. maybe a new enum would be better here?

Reply:
done